### ISSUE TITLE
Initial Results (Classification and NER tasks)

### ISSUE THREAD

**Issue by akash418** [malteos/finetune-evaluation-harness#3]
### Task: ```GERMEVAL_2018_OFFENSIVE_LANGUAGE``` : Type: Classification
Results:
- F-score (micro) 0.7517
- F-score (macro) 0.7123
- Accuracy 0.7517

```
By class:
              precision    recall  f1-score   support

       OTHER     0.7896    0.8502    0.8188      2330
     OFFENSE     0.6588    0.5607    0.6058      1202

    accuracy                         0.7517      3532
   macro avg     0.7242    0.7055    0.7123      3532
weighted avg     0.7451    0.7517    0.7463      3532

```

Settings:
- Transformer Document Embeddings
- Pooling: cls
- epochs: 100
- learning rate: 3e-5

### Task: ``` NER_GERMAN_LEGAL ```: Type NER
Results:
- F-score (micro) 0.9625
- F-score (macro) 0.9213
- Accuracy 0.9301

```
By class:
              precision    recall  f1-score   support

          GS     0.9779    0.9852    0.9815      1886
          RS     0.9760    0.9760    0.9760      1249
         GRT     0.9939    0.9879    0.9909       331
         LIT     0.9391    0.9544    0.9467       307
          VT     0.9386    0.9549    0.9466       288
         INN     0.9315    0.9231    0.9273       221
         PER     0.9579    0.9333    0.9455       195
          LD     1.0000    0.9869    0.9934       153
         EUN     0.9110    0.9433    0.9268       141
          RR     0.9922    1.0000    0.9961       127
         ORG     0.8000    0.7805    0.7901       123
          UN     0.9478    0.9646    0.9561       113
          VO     0.8696    0.9412    0.9040        85
          ST     0.9383    0.9870    0.9620        77
          VS     0.8209    0.8594    0.8397        64
         MRK     0.7895    0.8571    0.8219        35
         STR     0.7895    0.7500    0.7692        20
         LDS     0.7778    1.0000    0.8750        14
          AN     0.9167    1.0000    0.9565        11

   micro avg     0.9591    0.9660    0.9625      5440
   macro avg     0.9088    0.9360    0.9213      5440
weighted avg     0.9595    0.9660    0.9626      5440

```
Settings:
- Transformer Word Embeddings
- pooling_type: first_last
- epochs: 15
- learning_rate: 3e-5

**Comment by malteos** [comment]
Great job!

Few comments:

- Did you train the BERT model really for 100 epochs? Or was that max_epochs=100 and flair automatically stopped at one point?
- Please also run the evaluation with a "classifier-only" fine-tuning (transformer weights frozen). See https://github.com/flairNLP/flair/issues/2934
- Regarding GPT: What batch size / GPU memory did you use? Try to decrease the batch size until it fits to memory. And maybe also start with a smaller model, like https://huggingface.co/malteos/gpt2-wechsel-german-ds-meg

**Comment by malteos** [comment]
> I used 1 RTXA6000 GPU with a maximum of 89 GB of memory allocated to GPU, batch size 32, and hidden size 32. The best option is to try and decrease the batch size to 8 and try and see if it works. In worst case I will work with the smaller model.

You can even decrease the batch size to 1. Generally, please try to not always use the big GPUs on the cluster. For example, the RTX6000 should be totally sufficient.
